### PR TITLE
SAK-49345 Switch from Base62 to a variant of Base64

### PR DESCRIPTION
**Pull request by csev**
Hey @ern and @jonespm - You were right.  This commit switches to using Base64 URL Safe Encoding with a small adjustment to make double url encoding not a problem but without building our own Base62 implementation.  This in essence reverts the Base62 patch and replace it with an adjusted Base64 encoder.

### CODE REVIEW

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		int failcount = 0;
+
+		for (String input : testStrings) {
+			String base64UrlEncoded = java.util.Base64.getUrlEncoder().encodeToString(input.getBytes("UTF-8"));
+
+			// URL-encode the base64 URL encoded string
+			String urlEncoded = java.net.URLEncoder.encode(base64UrlEncoded, "UTF-8");
+
+			// Compare the original encoded string with the URL-encoded string
+			if ( ! base64UrlEncoded.equals(urlEncoded)) failcount ++;
```
Comment (ottenhoff) [inline basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:78]: you want all of these to fail?

if yes, should do an assertEquals for each one

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+public class Base64DoubleUrlEncodeSafeTest {
+
+	@Test
+	public void testEncodeAndDecode() {
+		String[] testStrings = {
+			"Hello, Base64DoubleUrlEncodeSafe!",
+			"Testing encoding and decoding.",
+			"12345",
+			"This is a longer string with more characters.",
+			"A tree 🌲 was here",
```
Comment (ottenhoff) [inline basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:24]: shouldn't one of these be a really complex url esp one with an underscore?

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import static org.junit.Assert.fail;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.Test;
+
+import org.tsugi.util.Base64DoubleUrlEncodeSafe;
+
+/* This code was written by ChatGPT with the following prompt
+ * Please write a unit test for a Base64DoubleUrlEncodeSafe class
+ */
```
Comment (ern) [inline basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:13]: useless comment

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+	@Test
+	public void testInvalidCharacters() {
+		// Test invalid characters in decoding
+		String invalidBase64DoubleUrlEncodeSafeString = "InvalidString$#@!";
+		try {
+			Base64DoubleUrlEncodeSafe.decode(invalidBase64DoubleUrlEncodeSafeString);
+			fail("Expected IllegalArgumentException for invalid Base64DoubleUrlEncodeSafe string");
+		} catch (Exception e) {
+			// Expected exception
```
Comment (ern) [inline basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:54]: you could also assert that an exception was caught, this is a good idea when an exception is expected

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java (hunk 1 above)
Comment (csev) [reply basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:78]: No - All I care about is that at least one fails - that is why I wrote it they way I did.

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java (hunk 2 above)
Comment (csev) [reply basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:24]: Good idea.

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java (hunk 3 above)
Comment (csev) [reply basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:13]: Yup.

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java (hunk 4 above)
Comment (csev) [reply basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:54]: OK

File: basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java (hunk 1 above)
Comment (csev) [reply basiclti/tsugi-util/src/test/org/tsugi/util/Base64DoubleUrlEncodeSafeTest.java:78]: They might not need padding so they don't all fail.

**Review by ern — APPROVED** [review]

**Pull request merged**